experimental/dds/tree2/src/feature-libraries/flex-tree/lazyNode.ts

```diff
@@ -287,43 +343,33 @@ export abstract class LazyTreeNode<TSchema extends TreeNodeSchema = TreeNodeSche
 						// Ugly casting workaround because I can't figure out how to make TS understand that in this case block
 						// the listener argument only needs to be an AnchorNode. Should go away if/when we make the listener signature
 						// for changing and subtreeChanging match the one for beforeChange and afterChange.
-						listener(anchorNode as unknown as AnchorNode & TreeEvent),
+						listener(anchorNode as unknown as AnchorNode & ITreeEvent),
 				);
 				return unsubscribeFromSubtreeChange;
 			}
 			case "beforeChange": {
-				const unsubscribeFromChildrenBeforeChange = this.#anchorNode.on(
-					"beforeChange",
-					(anchorNode: AnchorNode) => {
-						const treeNode = anchorNode.slots.get(lazyTreeSlot);
-						assert(
-							treeNode !== undefined,
-							0x7d3 /* tree node not found in anchor node slots */,
-						);
-						// Ugly casting workaround because I can't figure out how to make TS understand that in this case block
-						// the listener argument only needs to be a TreeEvent. Should go away if/when we make the listener signature
-						// for changing and subtreeChanging match the one for beforeChange and afterChange.
-						listener({ target: treeNode } as unknown as AnchorNode & TreeEvent);
-					},
-				);
-				return unsubscribeFromChildrenBeforeChange;
+				this.#listeners.set("beforeChange", [
+					...(this.#listeners.get("beforeChange") ?? []),
+					listener,
+				]);
```

Comment:
I'll see about using something simpler than a `Map` to avoid the verbosity here.